webscrambles/src/net/gnehzr/tnoodle/server/webscrambles/ScrambleRequest.java

```diff
@@ -622,21 +622,26 @@ private static void addFmcSolutionSheet(PdfWriter docWriter, Document doc, Scram
 
             offsetTop += fontSize + (int) (marginBottom * 2.8);
         }
-        fontSize = 15;
 
         rect = new Rectangle(competitorInfoLeft+padding, top-offsetTop, right-competitorInfoLeft, 100);
-        fitAndShowText(cb, translate("fmc.competitor", locale)+": __________________", bf, rect, fontSize, PdfContentByte.ALIGN_LEFT);
+        //fontSize = (int) (fitText(new Font(bf), translate("fmc.competitor", locale)+": __________________", rect, 15, false, 1f));
```

Comment:
This line of code ate the heap space, caused **OutOfMemory: java heap space** exception on my Ubuntu, so I removed it.

Reply:
This is a good start! I'm curious how did you came up with the `5` [here](https://github.com/thewca/tnoodle/pull/322/files#diff-5c6b26a54fa9a1a113fc535c6c81c91cR628).

Also, what about using rectangles to align WCA ID as well (just like you did in competitor)? This could reduce the need for [some variables](https://github.com/thewca/tnoodle/pull/322/files#diff-5c6b26a54fa9a1a113fc535c6c81c91cR635), make it easy for future updates and let it more readable.

Reply:
Well the number **5** was just my very first number coming into my head then. :smirk:
And for the idea of putting them in rectangles... Well, the **WCA ID** line has two parts with two different font sizes, so if we are going to put it in with **fitAndShowText**, we are going to add even more complicated codes (the **[fitAndShowText](https://github.com/thewca/tnoodle/pull/322/files#diff-5c6b26a54fa9a1a113fc535c6c81c91cR823)** thing is just putting the text into the lower-left corner of the rectangle).